### PR TITLE
[TODO] This method doesn't need to be public

### PR DESCRIPTION
**Pull request by quantumagi**
https://app.clickup.com/t/1vqt5h1

### CODE REVIEW

**Review by zeptin — DISMISSED** [review]

File: src/Stratis.Features.FederatedPeg/InputConsolidation/InputConsolidator.cs [hunk 1]
```diff
@@ -172,10 +172,8 @@ public ConsolidationSignatureResult CombineSignatures(Transaction incomingPartia
         /// </remarks>
         /// <param name="amount">The consolidation amount.</param>
         /// <returns>A list of consolidation transactions.</returns>
-        public List<ConsolidationTransaction> CreateRequiredConsolidationTransactions(Money amount)
+        private List<ConsolidationTransaction> CreateRequiredConsolidationTransactions(Money amount)
         {
-            // TODO: This method doesn't need to be public.
```
Comment (fassadlr) [inline src/Stratis.Features.FederatedPeg/InputConsolidation/InputConsolidator.cs:177]: I think this TODO isn't entirely valid as its OK to have a method `public` for unit testing purposes? Unless the whole consolidation process needs to be atomic. But then the unit test or design of the class needs to be revisited.

File: src/Stratis.Features.FederatedPeg/Stratis.Features.FederatedPeg.csproj [hunk 2]
```diff
@@ -22,6 +22,7 @@
     <ProjectReference Include="..\Stratis.Bitcoin.Features.Notifications\Stratis.Bitcoin.Features.Notifications.csproj" />
     <ProjectReference Include="..\Stratis.Bitcoin.Features.SmartContracts\Stratis.Bitcoin.Features.SmartContracts.csproj" />
     <ProjectReference Include="..\Stratis.Features.Collateral\Stratis.Features.Collateral.csproj" />
+    <ProjectReference Include="..\Stratis.Features.Diagnostic\Stratis.Features.Diagnostic.csproj" />
```
Comment (fassadlr) [inline src/Stratis.Features.FederatedPeg/Stratis.Features.FederatedPeg.csproj:25]: We shouldnt be adding this project reference. Im not sure what the purpose is of the `Diagnostic` project anymore anyway.

File: src/Stratis.Features.Diagnostic/Utils/ReflectionExtensionsion.cs [hunk 3]
```diff
@@ -54,5 +54,18 @@ public static T GetPrivateFieldValue<T>(this object obj, string fieldName)
...
         }
+
+        /// <summary>
+        /// Using reflection, calls a private method passing along the supplied parameters.
+        /// </summary>
+        /// <param name="obj">Object from where the Property Value is set</param>
+        /// <param name="methodName">Name of the method.</param>
+        /// <param name="parameters">Parameters for the method.</param>
+        /// <returns>The method return value.</returns>
+        public static object CallPrivateMethod(this object obj, string methodName, params object[] parameters)
```
Comment (fassadlr) [inline src/Stratis.Features.Diagnostic/Utils/ReflectionExtensionsion.cs:65]: If we want to keep that method, consider adding this moethod to `Stratis.Bitcoin`.

File: src/Stratis.Features.FederatedPeg/InputConsolidation/InputConsolidator.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg/InputConsolidation/InputConsolidator.cs:177]: Hmm I don't think we should need to change method visibility for tests, so for now maybe the PR will suffice until the design can perhaps be revisited.

**Review by fassadlr — APPROVED** [review]

File: src/Stratis.Features.Diagnostic/Utils/ReflectionExtensionsion.cs (hunk 3 above)
Comment (quantumagi) [reply src/Stratis.Features.Diagnostic/Utils/ReflectionExtensionsion.cs]: Done.

File: src/Stratis.Features.FederatedPeg/Stratis.Features.FederatedPeg.csproj (hunk 2 above)
Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg/Stratis.Features.FederatedPeg.csproj]: Removed.

**Pull request merged**